protocol/src/test/connection.test.ts

```diff
@@ -66,6 +71,91 @@ suite('Connection Tests', () => {
 		await clientConnection.sendRequest(InitializeRequest.type, init);
 		assert.ok(paramsCorrect, 'Parameters are transferred correctly');
 	});
+
+	test('Cancellation Tests', async () => {
```

Comment:
I would prefer having these tests in the cancellation package.

Reply:
cancellation package already have test. if you don't care protocol package testing custom cancellation strategy, then I can delete test here.

since the existing code base is structured in a way that each layer can be used standalone outside of LanguageClient, I added test to each layer that exposes CancellationStrategy to users